### PR TITLE
Move the required truncation of certificate expiration time to the GetExpiration method

### PR DESCRIPTION
**Pull request by steeling**
Move truncation into GetExpiration so GetExpiration correctly returns the time with monotonic clock truncated

### CODE REVIEW

File: pkg/certificate/manager.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ func (m *Manager) ShouldRotate(c *Certificate) bool {
 	// Round is called to truncate monotonic clock to the nearest second. This is done to avoid environments where the
 	// CPU clock may stop, resulting in a time measurement that differs significantly from the x509 timestamp.
 	// See https://github.com/openservicemesh/osm/issues/5000#issuecomment-1218539412 for more details.
-	expiration := c.GetExpiration().Round(0)
+	expiration := c.GetExpiration()
```
Comment (jsturtevant) [inline pkg/certificate/manager.go:154]: nit: should the comment get moved too?

**Review by jsturtevant — COMMENTED** [review]
lgtm

File: pkg/certificate/manager.go (hunk 1 above)
Comment (steeling) [reply pkg/certificate/manager.go:151]: yup good call!

**Review by jaellio — APPROVED** [review]

**Comment by keithmattix** [comment]
Looks good, but I'll approve when tests pass

**Review by keithmattix — APPROVED** [review]

**Pull request merged**